x-pack/legacy/plugins/siem/public/components/open_timeline/export_timeline/index.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useState, useCallback, Dispatch, SetStateAction } from 'react';
+import { keys } from 'lodash/fp';
+import { OpenTimelineResult } from '../types';
+import { KibanaServices } from '../../../lib/kibana';
+import { ExportSelectedData } from '../../generic_downloader';
+import { TIMELINE_EXPORT_URL } from '../../../../common/constants';
+
+export interface ExportTimeline {
+  enableDownloader: boolean;
+  setEnableDownloader: Dispatch<SetStateAction<boolean>>;
+  exportedIds:
+    | Array<{
+        timelineId: string | null | undefined;
+        pinnedEventIds: string[] | null | undefined;
+        noteIds: string[] | undefined;
+      }>
+    | undefined;
+  getExportedData: ExportSelectedData;
+}
+
+export const useExportTimeline = (
+  selectedItems: OpenTimelineResult[] | undefined
+): ExportTimeline => {
+  const [enableDownloader, setEnableDownloader] = useState(false);
+
+  const exportSelectedTimeline: ExportSelectedData = useCallback(
+    async ({
+      excludeExportDetails = false,
+      filename = `timelines_export.ndjson`,
+      ids = [],
+      signal,
+    }): Promise<Blob> => {
+      const body = ids.length > 0 ? JSON.stringify({ objects: ids }) : undefined;
```

Comment:
I think here we should take inspiration of what we are doing with detections engine/cases to create an API file and add the AbortController.

```
const fetchCasesStatus = useCallback(() => {
    let didCancel = false;
    const abortCtrl = new AbortController();
    const fetchData = async () => {
      setCasesStatusState({
        ...casesStatusState,
        isLoading: true,
      });
      try {
        const response = await getCasesStatus(abortCtrl.signal);
        if (!didCancel) {
          setCasesStatusState({
            ...response,
            isLoading: false,
            isError: false,
          });
        }
      } catch (error) {
        if (!didCancel) {
          errorToToaster({
            title: i18n.ERROR_TITLE,
            error: error.body && error.body.message ? new Error(error.body.message) : error,
            dispatchToaster,
          });
          setCasesStatusState({
            countClosedCases: 0,
            countOpenCases: 0,
            isLoading: false,
            isError: true,
          });
        }
      }
    };
    fetchData();
    return () => {
      didCancel = true;
      abortCtrl.abort();
    };
  }, [casesStatusState]);
```

Reply:
We have similar logic here:
https://github.com/elastic/kibana/blob/b7fd481009ae720f372d68222d80b328cb79700a/x-pack/legacy/plugins/siem/public/components/generic_downloader/index.tsx#L56-L107 

would we want to apply it again?

Reply:
Per our zoom chat, we decided to move this function into an isolated api file to avoid confusion.